core/src/Kokkos_CopyViews.hpp

```diff
@@ -1676,18 +1694,21 @@ void KOKKOS_INLINE_FUNCTION local_deep_copy(
   const size_t N = dst.extent(0) * dst.extent(1);
 
   if (dst.span_is_contiguous()) {
-    team.team_barrier();
-    local_deep_copy_contiguous(team, dst, value);
-    team.team_barrier();
-  } else {
-    team.team_barrier();
-    Kokkos::parallel_for(Kokkos::TeamVectorRange(team, N), [&](const int& i) {
-      int i0      = i % dst.extent(0);
-      int i1      = i / dst.extent(0);
-      dst(i0, i1) = value;
-    });
-    team.team_barrier();
+    if constexpr (std::is_same_v<decltype(dst.data()),
+                                 typename View<DT, DP...>::element_type*>) {
```

Comment:
Shouldn't this also use `!ViewTraits<DT, DP...>::impl_is_customized`?

Reply:
Actually yeah because we don't know what side effects an accessor could have. Hm I need to think about a test case for that.

Reply:
I thought more about it. Let's change that separately. I don't see anyone implementing something where this doesn't work for now. Even the accessors in the standard which come close to needing this (e.g. ScaledAccessor) wouldn't because they are read only access.

Reply:
Maybe add a comment.

Reply:
please definitely add a comment. Otherwise someone might read this later and get very nervous and confused